File: src/data/local/demoData.ts
/**
 * Datos de EJEMPLO (Fase 0, seccion 33). A proposito usa montos ficticios
 * distintos a los reales del usuario — el pedido explicito fue "no
 * inventes mis finanzas reales", asi que estos numeros son solo para
 * mostrar como se ve la app funcionando.
 */
import { addDays, parseISO, toISO } from '@/domain/dates';
import { calculateCreditCardCycle } from '@/domain/credit-card/cycle';
import type { Transaction } from '@/domain/types';
import { todayISO, nowISO } from '@/lib/todayISO';
import { db } from '../db';

function relativeDate(daysFromToday: number): string {
  return toISO(addDays(parseISO(todayISO()), daysFromToday));
}

export async function seedDemoTransactions(): Promise<void> {
  const now = nowISO();
  const tcPurchaseDate = relativeDate(-2);
  const tcCycle = calculateCreditCardCycle(tcPurchaseDate, 15, 2);

  const demo: Transaction[] = [
    {
      id: crypto.randomUUID(), type: 'income', concept: 'Ingreso de ejemplo',
      amount: 2_800_000, date: relativeDate(-5), categoryId: null, paymentMethodId: 'pm-debito',
      status: 'paid', quincenaKey: null, createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Arriendo (ejemplo)',
      amount: 1_800_000, date: relativeDate(-4), categoryId: 'cat-hogar', paymentMethodId: 'pm-debito',
      status: 'paid', quincenaKey: null, createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Suscripción streaming',
      amount: 24_900, date: relativeDate(-6), categoryId: 'cat-suscripciones', paymentMethodId: 'pm-debito',
      status: 'paid', quincenaKey: null, createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Mercado',
      amount: 145_000, date: relativeDate(-1), categoryId: 'cat-alimentacion', paymentMethodId: 'pm-debito',
      status: 'pending', quincenaKey: null, createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Gasolina',
      amount: 60_000, date: relativeDate(-3), categoryId: 'cat-transporte', paymentMethodId: 'pm-debito',
      status: 'paid', quincenaKey: null, createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Cine',
      amount: 38_000, date: relativeDate(0), categoryId: 'cat-entretenimiento', paymentMethodId: 'pm-tc',
      status: 'pending', quincenaKey: null,
      cycleCutoffDate: calculateCreditCardCycle(relativeDate(0), 15, 2).cycleCutoff,
      cyclePaymentDate: calculateCreditCardCycle(relativeDate(0), 15, 2).paymentDate,
      createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Zapatos (ejemplo)',
      amount: 210_000, date: tcPurchaseDate, categoryId: 'cat-compras', paymentMethodId: 'pm-tc',
      status: 'pending', quincenaKey: null,
      cycleCutoffDate: tcCycle.cycleCutoff, cyclePaymentDate: tcCycle.paymentDate,
      createdAt: now, updatedAt: now,
    },
    {
      id: crypto.randomUUID(), type: 'expense', concept: 'Gasto programado (ejemplo)',
      amount: 200_000, date: relativeDate(6), categoryId: 'cat-otros', paymentMethodId: 'pm-debito',
      status: 'scheduled', quincenaKey: null, createdAt: now, updatedAt: now,
    },
  ];

  await db.transactions.bulkPut(demo);
}

File: mobile/lib/data/demo_data.dart
/// Datos de ejemplo, para ver la app funcionando sin teclear veinte
/// movimientos. Las fechas son relativas a hoy, no fijas: con fechas fijas
/// el ejemplo se ve vacío apenas pasa el mes.
library;

import '../domain/credit_card/cycle.dart';
import '../domain/dates.dart';
import '../domain/types.dart';
import 'local_repository.dart';

IsoDate _relativa(int dias) => toIso(addDays(parseIso(todayIso()), dias));

Future<void> sembrarDemo(LocalRepository repo) async {
  final ahora = nowIso();
  final compraTc = _relativa(-2);
  final ciclo = calculateCreditCardCycle(compraTc, cutoffDay: 15, paymentDay: 2);

  final movimientos = <Transaction>[
    Transaction(
      id: 'demo-sueldo', type: TransactionType.income, concept: 'Sueldo', amount: 2800000,
      date: _relativa(-5), status: TransactionStatus.paid, paymentMethodId: 'pm-debito',
      createdAt: ahora, updatedAt: ahora,
    ),
    Transaction(
      id: 'demo-arriendo', type: TransactionType.expense, concept: 'Arriendo', amount: 1800000,
      date: _relativa(-4), status: TransactionStatus.paid, categoryId: 'cat-hogar',
      paymentMethodId: 'pm-debito', createdAt: ahora, updatedAt: ahora,
    ),
    Transaction(
      id: 'demo-mercado', type: TransactionType.expense, concept: 'Mercado', amount: 145000,
      date: _relativa(-1), status: TransactionStatus.pending, categoryId: 'cat-alimentacion',
      paymentMethodId: 'pm-debito', createdAt: ahora, updatedAt: ahora,
    ),
    Transaction(
      id: 'demo-gasolina', type: TransactionType.expense, concept: 'Gasolina', amount: 60000,
      date: _relativa(-3), status: TransactionStatus.paid, categoryId: 'cat-transporte',
      paymentMethodId: 'pm-debito', createdAt: ahora, updatedAt: ahora,
    ),
    Transaction(
      id: 'demo-zapatos', type: TransactionType.expense, concept: 'Zapatos', amount: 210000,
      date: compraTc, status: TransactionStatus.pending, categoryId: 'cat-compras',
      paymentMethodId: 'pm-tc', cycleCutoffDate: ciclo.cycleCutoff,
      cyclePaymentDate: ciclo.paymentDate, createdAt: ahora, updatedAt: ahora,
    ),
    Transaction(
      id: 'demo-servicios', type: TransactionType.expense, concept: 'Servicios', amount: 200000,
      date: _relativa(6), status: TransactionStatus.scheduled, categoryId: 'cat-servicios',
      paymentMethodId: 'pm-debito', createdAt: ahora, updatedAt: ahora,
    ),
    Transaction(
      id: 'demo-freelance', type: TransactionType.income, concept: 'Freelance', amount: 700000,
      date: _relativa(4), status: TransactionStatus.pending, paymentMethodId: 'pm-debito',
      createdAt: ahora, updatedAt: ahora,
    ),
  ];

  await repo.saveTransactions(movimientos);
}
